src/libs/auth.test.js

```diff
@@ -0,0 +1,144 @@
+      currentVersion: '9',
+      userAcceptedVersion: '0',
+      userCanUseTerra: true,
+      showTosPopup: true
+    })
+  })
+
+  it('does not give access when the user has not accepted the latest ToS version and the grace period is disabled', () => {
+    const tosDetails = {
+      ...tosDetailsDefault,
+      currentVersion: '1',
+      userAcceptedVersion: '0',
+      isGracePeriodEnabled: false
+    }
+
+    expect(parseToSDetails(tosDetails)).toEqual({
+      isGracePeriodEnabled: false,
+      currentVersion: '1',
+      userAcceptedVersion: '0',
+      userCanUseTerra: false,
+      showTosPopup: true
+    })
+  })
+
+  it('does not give access when the user has never accepted any version of the ToS', () => {
+    const tosDetails = {
+      ...tosDetailsDefault,
+      userAcceptedVersion: undefined
+    }
+
+    const expectedResult = {
+      isGracePeriodEnabled: true,
+      currentVersion: '1',
+      userAcceptedVersion: undefined,
+      userCanUseTerra: false,
+      showTosPopup: true
+    }
+
+    // Assertion for when `userAcceptedVersion` is undefined
+    expect(parseToSDetails(tosDetails)).toEqual(expectedResult)
+
+    // Assertion for when key `userAcceptedVersion` is not present in object
+    expect(parseToSDetails({
+      currentVersion: '1',
+      isEnabled: true,
+      isGracePeriodEnabled: true
+    })).toEqual(expectedResult)
+  })
+})
+
+describe('Terms of Service Detail Parser sets ToS pop-up', () => {
+  it('to show if the user has accepted a version of the ToS that is not the latest version and the grace period is enabled', () => {
+    expect(parseToSDetails({
+      isGracePeriodEnabled: true,
+      isEnabled: true,
+      currentVersion: '1',
+      userAcceptedVersion: '0'
+    })).toEqual({
+      isGracePeriodEnabled: true,
+      currentVersion: '1',
+      userAcceptedVersion: '0',
+      userCanUseTerra: true,
+      showTosPopup: true // this is the field we care about for this test
+    })
+  })
+
+  it('to show if the user has accepted a version of the ToS that is not the latest version and the grace period is disabled', () => {
+    expect(parseToSDetails({
+      isGracePeriodEnabled: false,
+      isEnabled: true,
+      currentVersion: '1',
+      userAcceptedVersion: '0'
+    })).toEqual({
+      isGracePeriodEnabled: false,
+      currentVersion: '1',
+      userAcceptedVersion: '0',
+      userCanUseTerra: false,
+      showTosPopup: true // this is the field we care about for this test
+    })
+  })
+
+  it('to show if the user has never accepted a version of the ToS', () => {
+    expect(parseToSDetails({
+      isGracePeriodEnabled: false,
+      isEnabled: true,
+      currentVersion: '1',
+      userAcceptedVersion: undefined
+    })).toEqual({
+      isGracePeriodEnabled: false,
+      currentVersion: '1',
+      userAcceptedVersion: undefined,
+      userCanUseTerra: false,
+      showTosPopup: true // this is the field we care about for this test
+    })
+  })
+})
```

Comment:
great set of tests!

While I don't want to holdup the fix for a prod issue, would be good to follow-up with a PR that adds unit tests to the other files being changed in this PR.  I know it's always hard to write the first unit test on a module, especially if it's a large visual component... but these all look manageable to at least do a focused pass that unit tests the key bits this PR is concerned about.